packages/salesforcedx-vscode-core/test/vscode-integration/commands/forceCreateManifest.test.ts

```diff
@@ -0,0 +1,175 @@
+        }
+      });
+      env.stub(fs, 'existsSync')
+        .onFirstCall()
+        .returns(true)
+        .onSecondCall()
+        .returns(false);
+      env.stub(fs, 'writeFileSync').returns(undefined);
+      const packageName = 'package' + generateRandomSuffix() + '.xml';
+      inputBoxSpy.onCall(0).returns(packageName);
+      await forceCreateManifest(uri1, [uri1, uri2]);
+
+      expect(openDocSpy.calledOnce).to.equal(true);
+    });
+
+    it('Should create but not save manifest if cancelled', async () => {
+      const writeFileSpy = env.spy(fs, 'writeFileSync');
+      const packageXML = util.format(manifest, `<member>${class1}</member>\n<member>${class2}</member>\n`);
+      env.stub(ComponentSet, 'fromSource').returns({
+        getPackageXml: () => {
+          return packageXML;
+        }
+      });
+      env.stub(fs, 'existsSync')
+        .onFirstCall()
+        .returns(true)
+        .onSecondCall()
+        .returns(false);
+      inputBoxSpy.onCall(0).returns(undefined);
+      await forceCreateManifest(uri1, [uri1, uri2]);
+
+      expect(openDocSpy.calledOnce).to.equal(true);
+      expect(writeFileSpy.called).to.equal(false);
+    });
+
+  });
+
+  describe('Exception Handling', () => {
+
+    let env: sinon.SinonSandbox;
+    let openDocSpy: sinon.SinonSpy;
+    let inputBoxSpy: sinon.SinonStub;
+
+    before(() => {
+      env = createSandbox();
+    });
+
+    beforeEach(() => {
+      inputBoxSpy = env.stub(vscode.window, 'showInputBox');
+      openDocSpy = env.spy(vscode.workspace, 'openTextDocument');
+    });
+
+    afterEach(() => {
+      env.restore();
+    });
+
+    it('Should handle exception while creating component set', async () => {
+      env.stub(ComponentSet, 'fromSource').throws(
+        new Error(nls.localize('error_creating_packagexml'))
+      );
+      let exceptionThrown = false;
+      try {
+        await forceCreateManifest(uri1, [uri2]);
+        fail('Should have thrown exception');
+      } catch (e) {
+        expect(e.message).to.contain('package.xml');
+        exceptionThrown = true;
+      }
+      expect(exceptionThrown).to.equal(true);
+      expect(openDocSpy.called).to.equal(false);
+    });
+
+    it('Should enforce unique manifest names', async () => {
+      const packageXML = util.format(manifest, `<member>${class1}</member>`);
+      env.stub(ComponentSet, 'fromSource').returns({
+        getPackageXml: () => {
+          return packageXML;
+        }
+      });
+      env.stub(fs, 'existsSync').returns(true);
+      const fileName = 'duplicatePackageName';
+      inputBoxSpy.resolves(fileName);
+
+      let exceptionThrown = false;
+      try {
+        await forceCreateManifest(uri1, undefined);
+      } catch (e) {
+        exceptionThrown = true;
+        expect(e.message).to.contain(fileName);
+      }
+      expect(exceptionThrown).to.equal(true);
+      expect(openDocSpy.called).to.equal(false);
+    });
+  });
+
+});
+
+function generateRandomSuffix() {
```

Comment:
nit: I personally prefer to default to arrow function assigned to constants pretty much always.  It removes the need to consider the good ol `scope/this` in which the function is executed in. It's one of those things where it's usually not an issue, but when it is it totally sucks and is hard to debug.

Reply:
I always thought this was just a style preference. Looks like there is more to this. Can we take it up in one of the elevensies, please?

Reply:
@jag-sfdc totally.

Reply:
we're still discussing as a team. Can follow up later after we decide on a coding convention.